types/transaction.go

```diff
@@ -24,10 +24,26 @@ type Transaction struct {
 	size atomic.Value
 }
 
+// IsContractCreation checks if tx is contract creation
 func (t *Transaction) IsContractCreation() bool {
 	return t.To == nil
 }
 
+// CanDeployContract checks if address can deploy smart contract
+func (t *Transaction) CanDeployContract(whitelist []Address) bool {
+	// If whitelist is empty anyone can deploy
+	if len(whitelist) == 0 {
+		return true
+	}
+
+	// Only addresses which exists in whitelist can deploy
+	if t.From.ExistsIn(whitelist) {
+		return true
+	}
+
+	return false
+}
+
```

Comment:
Why not define this as a private helper in the pool, and not as an exported method of the `Transaction` type?

Reply:
You can also simplify this method:
```go
    return len(whitelist) == 0 || t.From.ExistsIn(whitelist)
```

Reply:
For a 2 resons. First i can see validating this in the future maybe even in executor. Second it feels more related to the transaction then to the pool. 
But i can change this

Reply:
@dbrajovic @Kourin1996 what do you think?

Reply:
> First i can see validating this in the future maybe even in executor

Please let's not discuss how we want to implement  _helper_ methods in the future. We'll move it if it needs to become more general. 
How is it related to the transaction more? You are literally doing a validation based off of information in the pool, and not in the transaction itself

Reply:
You really do love helper methods 😍 4ebc2ec3e4b406979222489c176a8c3339b13068